motr/client.h

```diff
@@ -591,6 +591,43 @@ enum m0_entity_type {
 	M0_ET_IDX
 } M0_XCA_ENUM;
 
+/**
+ * Flags passed to m0_entitiy_create(), m0_entity_open() to specify applications's 
+ * behaviour.
+ */
+ enum m0_entity_flags {
+	/**
+	 * During create if this flag is set in entity->en_flags, that means
+	 * application has capability to store meta-data and hence pver and
+	 * lid can be store in  application's meta-data.
+	 * Before application call to m0_entity_create/open(), applications is
+	 * expected to set obj>ob_entity->en_flags != M0_ENF_META, so when
+	 * m0_entity_create() returns to application, pool version and layout id
+	 * will be available to application into obj->ob_attr.oa_pver and
+	 * obj->ob_attr.oa_lid respectively and can be store into applications
+	 * meta-data.
+	 *
+	 * For example:
+	 * Create workflow would be like:
+	 *   obj->ob_entity.en_flags |= M0_ENF_META;
+	 *   m0_entity_create((NULL, &obj.ob_entity, &ops[0]);
+	 *   //  Save the returned pool version and lid into app_meta_data
+	 *   app_meta_data.pver = obj->ob_attr.oa_pver;
+	 *   app_meta_data.lid = obj->ob_attr.oa_lid;
```

Comment:
I didn't see where the "obj->ob_attr.oa_lid" is used.
Is it in another PR?

Reply:
Actually this requirement came up like we need to send pver and layout id to application who has capability to store meta-data(In current case it is s3) and same pool version, layout_id can be used to read the object back without needing to get it again from server. 

But actually layout id is already getting set in m0_obj_init() before create/open entity, 
with the layout id received in m0_obj_init() or to the default. (This code is present in main since beginning)
         obj->ob_attr.oa_layout_id = layout_id ?: M0_DEFAULT_LAYOUT_ID;

This patch really don't have any changes related to layout_id, but as per design application is expected to pass layout id and pool version in object attributes, to read object. So I think it is better to keep comment as it is. But can be removed if not not seems relevant or confusing, what do you suggest ?

Reply:
OK